src/EditorFeatures/CSharpTest/EventHookup/EventHookupCommandHandlerTests.cs

```diff
@@ -1080,6 +1080,31 @@ void CurrentDomain_UnhandledException(object sender, System.UnhandledExceptionEv
             testState.AssertCodeIs(expectedCode);
         }
 
+        [WpfFact, Trait(Traits.Feature, Traits.Features.EventHookup)]
+        [WorkItem(59935, "https://github.com/dotnet/roslyn/issues/59935")]
+        public async Task HandlerName_EventInGenericClass()
+        {
+            var markup = @"
+using System;
+
+class C
+{
+    void M()
+    {
+        Generic&lt;int&gt;.MyEvent +$$
```

Comment:
Can you add a test for `global::Generic<int>.MyEvent`? I'm expecting it to fail.

Reply:
For that, I suggest using this extension method

https://github.com/dotnet/roslyn/blob/31d7f1f7da00ffc8a6aeaf6de97159ee2caa0bae/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs#L679

Reply:
Yes, it failed both for generic and non-generic. I added more tests and fix that